ddl/ddl_api.go

```diff
@@ -205,6 +205,11 @@ func setCharsetCollationFlenDecimal(tp *types.FieldType) error {
 	defaultFlen, defaultDecimal := mysql.GetDefaultFieldLengthAndDecimal(tp.Tp)
 	if tp.Flen == types.UnspecifiedLength {
 		tp.Flen = defaultFlen
+		if mysql.HasUnsignedFlag(tp.Flag) && tp.Tp != mysql.TypeLonglong && mysql.IsIntegerType(tp.Tp) {
+			// Issue #4684: unsigned integer(except bigint) is one fewer flen than signed integer
```

Comment:
why leave out `bigint` ?

Reply:
The flen of the unsigned integer(except bigint) is 1 digit shorter than signed integer because ...